homeassistant/components/smartthings/cover.py

```diff
@@ -0,0 +1,146 @@
+"""Support for covers through the SmartThings cloud API."""
+from typing import Optional, Sequence
+
+from homeassistant.components.cover import (
+    ATTR_POSITION, DEVICE_CLASS_DOOR, DEVICE_CLASS_GARAGE, DEVICE_CLASS_SHADE,
+    DOMAIN as COVER_DOMAIN, STATE_CLOSED, STATE_CLOSING, STATE_OPEN,
+    STATE_OPENING, SUPPORT_CLOSE, SUPPORT_OPEN, SUPPORT_SET_POSITION,
+    CoverDevice)
+from homeassistant.const import ATTR_BATTERY_LEVEL
+
+from . import SmartThingsEntity
+from .const import DATA_BROKERS, DOMAIN
+
+DEPENDENCIES = ['smartthings']
+
+VALUE_TO_STATE = {
+    'closed': STATE_CLOSED,
+    'closing': STATE_CLOSING,
+    'open': STATE_OPEN,
+    'opening': STATE_OPENING,
+    'partially open': STATE_OPEN,
+    'unknown': None
+}
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Platform uses config entry setup."""
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add covers for a config entry."""
+    broker = hass.data[DOMAIN][DATA_BROKERS][config_entry.entry_id]
+    async_add_entities(
+        [SmartThingsCover(device) for device in broker.devices.values()
+         if broker.any_assigned(device.device_id, COVER_DOMAIN)], True)
+
+
+def get_capabilities(capabilities: Sequence[str]) -> Optional[Sequence[str]]:
+    """Return all capabilities supported if minimum required are present."""
+    from pysmartthings import Capability
+
+    min_required = [
+        Capability.door_control,
+        Capability.garage_door_control,
+        Capability.window_shade
+    ]
+    supported = min_required.copy()
```

Comment:
Why do we need to copy?

Reply:
Because we're creating another list (and adding to it) in the next few lines.  `min_required` is what the device must have, then `supported` contains the additional ones it consumes.  I could extend the original list in the return statement, but though this was more understandable.

Reply:
I think it's ok to reuse the first list, since we create it on each call of this function.